### ISSUE TITLE
RePostPayment fails with System.InvalidOperationException: Headers are read-only, response has already started.

### ISSUE THREAD

**Issue by alinmircea** [nopSolutions/nopCommerce#7314]
nopCommerce version:  4.6+

Steps to reproduce the problem: 

Calling RePostPayment fails with "System.InvalidOperationException: Headers are read-only, response has already started." 
         at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpHeaders.ThrowHeadersReadOnlyException()
         at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponseHeaders.Microsoft.AspNetCore.Http.IHeaderDictionary.set_ContentType(StringValues value)
         at Microsoft.AspNetCore.Http.DefaultHttpResponse.set_ContentType(String value)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ContentResultExecutor.ExecuteAsync(ActionContext context, ContentResult result)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|30_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeResultFilters()
      --- End of stack trace from previous location ---

https://github.com/nopSolutions/nopCommerce/blob/020f7571c2f37beedc33c8bfb04ab53565623c1b/src/Presentation/Nop.Web/Controllers/OrderController.cs#L237

Will write the form to the body.

https://github.com/nopSolutions/nopCommerce/blob/020f7571c2f37beedc33c8bfb04ab53565623c1b/src/Presentation/Nop.Web/Controllers/OrderController.cs#L242

Will try to set Content Type header because of Content("Redirected").

----------------------------

Consider  "return new EmptyResult()" instead of "Content("Redirected")", along with adding a <noscript> wrapped text "please enable javascript" or similar, to RemotePost

https://github.com/nopSolutions/nopCommerce/blob/020f7571c2f37beedc33c8bfb04ab53565623c1b/src/Presentation/Nop.Web.Framework/RemotePost.cs#L115

**Comment by AndreiMaz** [comment]
@alinmircea What payment method (plugin) are you using?

**Comment by alinmircea** [comment]
Custom implementation for https://netopia-payments.com/

PaymentMethodType.Redirection plugin, where PostProcessPaymentAsync(PostProcessPaymentRequest postProcessPaymentRequest) does:

```
RemotePost post = new RemotePost(_httpContextAccessor, _webHelper);
post.NewInputForEachValue = true;
post.Url = messageEndpoint;
post.Add("data", encrypt.EncryptedData);
post.Add("env_key", encrypt.EnvelopeKey);
post.Post();
```

**Comment by skoshelev** [comment]
Closed #7314
